src/Common/src/TypeSystem/Interop/IL/Marshaller.cs

```diff
@@ -1610,6 +1610,29 @@ protected override void EmitCleanupManaged(ILCodeStream codeStream)
 
     class SafeHandleMarshaller : ReferenceMarshaller
     {
+        protected override void AllocNativeToManaged(ILCodeStream codeStream)
+        {
+            var ctor = ManagedType.GetParameterlessConstructor();
+            if (ctor == null)
+            {
+                var emitter = _ilCodeStreams.Emitter;
+
+                MethodSignature ctorSignature = new MethodSignature(0, 0, Context.GetWellKnownType(WellKnownType.Void),
+                      new TypeDesc[] { Context.GetWellKnownType(WellKnownType.String) });
+                MethodDesc exceptionCtor = InteropTypes.GetMissingMemberException(Context).GetKnownMethod(".ctor", ctorSignature);
+
+                MetadataType type = ManagedType as MetadataType;
+                string name = type != null ?  type.Name : "Subclass of SafeHandle";
```

Comment:
When is `ManagedType` not a `MetadataType`? If we know `ManagedType` derives from `SafeHandle`, it must be a `MetadataType`.

If this is just some defensive programming, let's not do that. I prefer the compiler to crash if there's something wrong instead of silently generating bad code. It's easier to troubleshoot.